crypto/fipsmodule/cipher/cipher.c

```diff
@@ -424,6 +427,8 @@ int EVP_DecryptUpdate(EVP_CIPHER_CTX *ctx, uint8_t *out, int *out_len,
     *out_len += b;
   }
 
+out:
+  EVP_Cipher_verify_service_indicator(ctx);
```

Comment:
It's my fault for not testing this, but `EVP_EncryptUpdate` and `EVP_DecryptUpdate` shouldn't be able to update the state of the service indicator. Our discussions behind this were that the "service" hasn't finished during `EVP_*Update`, and one could only know if the entire "service" operation had finished when the user called `EVP_*Final`. 
That being said, maybe we should add a test similar to the code below in `DoCipherFinal` (`crypto/fipsmodule/service_indicator/service_indicator_test.cc`)
https://github.com/awslabs/aws-lc/blob/07f270f995e6ba1dac141d5349d07bfa81ccb623/crypto/fipsmodule/service_indicator/service_indicator_test.cc#L825-L826

Reply:
I've backed out the `EncryptUpdate`/`DecryptUpdate` changes and (I think) modified `DoCipherFinal()` as you've suggested.